pkg/client/core/service_application_client.go

```diff
@@ -86,14 +87,120 @@ func (s *applicationService) ScaffoldApplication(ctx context.Context, opts *clie
 	return nil
 }
 
+func getApplicationOverlayClusters(fs *afero.Afero, absoluteApplicationDir string) ([]string, error) {
+	absoluteApplicationOverlaysDirectory := path.Join(absoluteApplicationDir, paths.DefaultApplicationOverlayDir)
+
+	clusters := make([]string, 0)
+
+	err := fs.Walk(absoluteApplicationOverlaysDirectory, func(currentPath string, info fsPkg.FileInfo, err error) error {
+		if err != nil {
+			if stderrors.Is(err, os.ErrNotExist) {
+				return nil
+			}
+
+			return err
+		}
+
+		if contains(recognizedKustomizationFileNames(), path.Base(currentPath)) {
+			dir, _ := path.Split(currentPath)
+
+			clusters = append(clusters, path.Base(dir))
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("walking overlays directory: %w", err)
+	}
+
+	return clusters, nil
+}
+
+func getClusterApplications(fs *afero.Afero, absoluteRepositoryRootDir string, cluster v1alpha1.Cluster) ([]string, error) {
+	absoluteApplicationsDir := path.Join(absoluteRepositoryRootDir, paths.GetRelativeArgoCDApplicationsDir(cluster))
+
+	applications := make([]string, 0)
+
+	items, err := fs.ReadDir(absoluteApplicationsDir)
+	if err != nil {
+		if stderrors.Is(err, os.ErrNotExist) {
+			return applications, nil
+		}
+
+		return nil, fmt.Errorf("listing applications directory: %w", err)
+	}
+
+	for _, item := range items {
+		valid, err := argocd.IsArgoCDApplication(fs, path.Join(absoluteApplicationsDir, item.Name()))
+		if err != nil {
+			return nil, fmt.Errorf("checking if ArgoCD application: %w", err)
+		}
+
+		if valid {
+			applications = append(applications, strings.Replace(item.Name(), path.Ext(item.Name()), "", 1))
+		}
+	}
+
+	return applications, nil
+}
+
+// AmountAssociatedClusters knows how to count the number of associated clusters a specific application has. A cluster is
+// classified as associated iff the cluster has an ArgoCD application referencing the app and the app has an overlay folder
+// referencing the cluster.
+func AmountAssociatedClusters(fs *afero.Afero, absoluteRepositoryRootDir string, clusterContext v1alpha1.Cluster, app v1alpha1.Application) (int, error) {
```

Comment:
Consider: A common way I've seen of naming functions and variables is using the word "count", for instance as a noun
* `GetAssoicatedClustersCount`

or a verb:
* `CountAssociatedClusters`